### PR TITLE
`Communication`: Fix code of conduct to only appear if it has not been accepted

### PR DESCRIPTION
**Pull request by rstief**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Fix Issue 6 of [7.0.0 Milestone Testing](https://confluence.ase.in.tum.de/display/ArTEMiS/Testing+-+7.0.0+Milestone)

### Description
Fixes the loading of `isCodeOfConductAccepted` so the code of conduct only shows if it has not been accepted.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course -> Communication
3. Accept the Code of Conduct
4. Reload
5. See that the code of conduct does not flash for a split second

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
unchanged

### CODE REVIEW

**Review by Jan-Thurner — DISMISSED** [review]
code 👍

**Review by laurenzfb — DISMISSED** [review]
looks good 🪨

**Review by milljoniaer — APPROVED** [review]
works like a charm 👍

**Review by JohannesWt — APPROVED** [review]
Looks good to me

**Review by MaximilianAnzinger — DISMISSED** [review]
Clean fix, thank you 👍

File: src/main/webapp/app/overview/course-conversations/course-conversations.component.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <jhi-loading-indicator-container [isLoading]="isLoading">
-    @if (course && !isCodeOfConductAccepted) {
+    <!-- only display after isCodeOfConductAccepted is loaded ( !== undefined) -->
+    @if (course && isCodeOfConductAccepted !== undefined && !isCodeOfConductAccepted) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/overview/course-conversations/course-conversations.component.html:3]:
```suggestion
    @if (course && isCodeOfConductAccepted === false) {
```

**Review by Strohgelaender — DISMISSED** [review]

**Review by JohannesWt — APPROVED** [review]
Code looks good

**Review by Strohgelaender — APPROVED** [review]
re-approve

**Pull request merged**